lib/metrics.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Cloudex.Metrics do
+  require Logger
+  import Telemetry.Metrics
+
+  def setup do
+    events = [
+      [:cloudex, :upload_large, :success],
+      [:cloudex, :upload_chunk, :success],
+      [:cloudex, :upload_large, :failure],
+      [:cloudex, :upload_chunk, :failure]
+    ]
+  
+    :telemetry.attach_many("cloudex-metrics", events, &handle_event/4, nil)
+  end
+
+  def handle_event([:cloudex, :upload_large, :success], measurements, metadata, _config) do
+    Logger.info inspect(measurements)
+    Logger.info inspect(metadata)
+    counter("cloudex.upload_large.success")
```

Comment:
just put some functions for the time being don't know what all we should have here, counter and summary made sense to me based on reading the `telemetry_metrics` docs